mesa_geo/geospace.py

```diff
@@ -48,19 +49,65 @@ def __init__(self, crs="epsg:3857"):
         self.idx = index.Index()
         self.idx.agents = {}
 
-    def add_agents(self, agents):
+    @property
+    def crs(self):
+        """
+        Return the coordinate reference system of the GeoSpace.
+        """
+        return self._crs
+
+    @property
+    def transformer(self):
+        """
+        Return the pyproj.Transformer that transforms the GeoSpace into
+        epsg:4326. Mainly used for GeoJSON serialization.
+        """
+        return self._transformer
+
+    def add_agents(self, agents, auto_convert_crs=False):
         """Add a list of GeoAgents to the Geospace.
 
         GeoAgents must have a geometry attribute. This function may also be called
-        with a single GeoAgent."""
+        with a single GeoAgent.
+
+        Args:
+            agents: List of GeoAgents, or a single GeoAgent, to be added into GeoSpace.
+            auto_convert_crs: Whether to automatically convert GeoAgent of different
+                crs into the crs of the GeoSpace. Default to False.
+
+        Raises:
+            ValueError: If GeoAgent of different crs is added into the GeoSpace, while
+                `self.auto_convert_crs` is set to False.
+        """
         if isinstance(agents, GeoAgent):
             agent = agents
             if hasattr(agent, "geometry"):
+                if not self.crs.is_exact_same(agent.crs):
+                    if auto_convert_crs:
+                        agent.to_crs(self.crs)
+                    else:
+                        raise ValueError(
+                            f"Unable to add agents due to inconsistent crs: {agent.__class__.__name__}"
```

Comment:
It's hard to tell which crs is the agent's or the space's if you were to render the error message this way.

Reply:
Thanks for the feedback. I'll get some sleep and rephrase this tomorrow : )